src/gui/addnewtorrentdialog.cpp

```diff
@@ -621,32 +636,63 @@ void AddNewTorrentDialog::displayContentTreeMenu(const QPoint &)
     }
 }
 
+void AddNewTorrentDialog::setAddTorrentParams(const BitTorrent::AddTorrentParams &inParams)
+{
+    torrentParams = inParams;
+    
+    if (!torrentParams.savePath.isEmpty())
+        setSavePath(torrentParams.savePath);
+    
+    if (!torrentParams.category.isEmpty()) {
+        int index = ui->categoryComboBox->findText(torrentParams.category);
+        
+        if (index != -1) {
+            ui->categoryComboBox->setCurrentIndex(index);
+        }
+        else {
+            ui->categoryComboBox->addItem(torrentParams.category);
```

Comment:
I don't think this relation is correct now. In the GUI user actions are explicit, they know that the new category is created, but here this is done implicitly. To make these two cases equivalent, we should add to the command line help and note that category will be created if it does not exist.